projects/libyuv/libyuv_fuzzer.cc

```diff
@@ -0,0 +1,80 @@
+/* # Copyright 2020 Google Inc. */
+/* # */
+/* # Licensed under the Apache License, Version 2.0 (the "License"); */
+/* # you may not use this file except in compliance with the License. */
+/* # You may obtain a copy of the License at */
+/* # */
+/* #      http://www.apache.org/licenses/LICENSE-2.0 */
+/* # */
+/* # Unless required by applicable law or agreed to in writing, software */
+/* # distributed under the License is distributed on an "AS IS" BASIS, */
+/* # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. */
+/* # See the License for the specific language governing permissions and */
+/* # limitations under the License. */
+/* # */
+/* ################################################################################ */
+
+#include <iostream>
+#include <vector>
+#include <cmath>
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include "libyuv.h"
+#include "fuzz_common.h"
+
+using namespace std;
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+
+    FuzzedDataProvider fuzzed_data(data, size);
+
+    int range_size = sqrt((static_cast<int>(size) + 1) / 2);
+    if (range_size == 0)
+    {
+        libyuv::I420ToNV21(data, 0, data, 0, data, 0, nullptr, 0, nullptr, 0, 0, 0);
+        return 0;
+    }
+
+    size_t width = fuzzed_data.ConsumeIntegralInRange(1, range_size);
+    size_t height = fuzzed_data.ConsumeIntegralInRange(1, range_size);
+    bool cpu_flag = fuzzed_data.ConsumeBool();
+
+    write_conf(width, height, cpu_flag);
+
+    if (cpu_flag == 1)
+    {
+        //enable all cpu specific optimizations.
+        libyuv::MaskCpuFlags(-1);
+    }
+    else
+    {
+        //disable all cpu specific optimizations.
+        libyuv::MaskCpuFlags(1);
+    }
+
+    int kPixels = width * height;
+    int kHalfPixels = ((width + 1) / 2) * ((height + 1) / 2);
+
+    std::vector<uint8_t> first_part =
+        fuzzed_data.ConsumeBytes<uint8_t>(kPixels);
+    std::vector<uint8_t> second_part =
+        fuzzed_data.ConsumeBytes<uint8_t>(kHalfPixels);
+    std::vector<uint8_t> third_part =
+        fuzzed_data.ConsumeRemainingBytes<uint8_t>();
+
+    uint8_t *dst_data = new uint8_t [size];
+
+    uint8_t *dst_y_data = dst_data;
+    uint8_t *dst_uv_data = dst_data + kPixels;
+
+    const uint8_t *src_y_data = first_part.data();
+    const uint8_t *src_u_data = second_part.data();
+    const uint8_t *src_v_data = third_part.data();
+
+    libyuv::I420ToNV21(src_y_data, width, src_u_data, (width + 1) / 2, src_v_data, (width + 1) / 2, dst_y_data, width, dst_uv_data, ((width + 1) / 2) * 2, width, height);
```

Comment:
this tests the simple case where width == stride.
the function will notice this and I420ToNV21 will treat the each plane as a single row.
e.g. the Y plane of I420 and NV21 are actually the same.  If width = stride it will change the parameters internally to 0 for stride and width * height, 1 for width and height, making a single call to memcpy() for the entire image, testing very different code, even if the width was odd.
ideally you would test where width is less than stride, forcing the code to do 1 row at a time.
libyuv functions also allow a negative stride, which means the image will be inverted, but you will need to adjust the starting pointer if you care to test that.
the stride could be randomized to a value less than width, but the results of the image would be unpredictable and not all source bytes would be consumed.
the width could be randomized to be less than stride.  it buffer size is determined by stride but you can process a subregion.